tests/drop_counters/test_drop_counters.py

```diff
@@ -0,0 +1,383 @@
+import os
+import json
+
+pytestmark = [
+    pytest.mark.disable_loganalyzer  # disable automatic loganalyzer
+]
+
+logger = logging.getLogger(__name__)
+
+IP_DST = "10.0.0.59"
+IP_SRC = "1.1.1.1"
+TCP_SPORT = 1234
+TCP_DPORT = 4321
+PKT_NUMBER = 1000
+
+# Discard key from 'portstat -j' CLI command output
+L2_DISCARD_KEY = "RX_DRP"
+L3_DISCARD_KEY = "RX_ERR"
+# CLI commands to obtain drop counters
+GET_L2_COUNTERS = "portstat -j"
+GET_L3_COUNTERS = "intfstat -j"
+
+
+@pytest.fixture(scope="module")
+def setup(duthost, testbed):
+    """
+    Setup fixture for collecting PortChannel, VLAN and RIF port members.
+    @return: Dictionary with keys:
+        port_channel_members, vlan_members, rif_members, dut_to_ptf_port_map, combined_drop_counter
+        Note: if 'combined_drop_counter' is True, platform has common counter for L2 and L3 discards.
+              To get this counter - call 'show interfaces counters' CLI comamnd and check 'RX_DRP' column for
+              specified port.
+    """
+    port_channel_members = {}
+    vlan_members = {}
+    rif_members = []
+    combined_drop_counter = False
+
+    # Gather ansible facts
+    mg_facts = duthost.minigraph_facts(host=duthost.hostname)['ansible_facts']
+
+    for port_channel, interfaces in mg_facts['minigraph_portchannels'].items():
+        for iface in interfaces["members"]:
+            port_channel_members[iface] = port_channel
+
+    for vlan_id in mg_facts["minigraph_vlans"]:
+        for iface in mg_facts["minigraph_vlans"][vlan_id]["members"]:
+            vlan_members[iface] = vlan_id
+
+    rif_members = {item["attachto"]: item["attachto"] for item in mg_facts["minigraph_interfaces"]}
+
+    # Get info whether L2 and L3 drop counters are linked
+    base_dir = os.path.dirname(os.path.realpath(__file__))
+    with open(os.path.join(base_dir, "combined_drop_counters.yml")) as stream:
+        regexps = yaml.safe_load(stream)
+        if regexps:
+            for item in regexps:
+                if re.match(item, duthost.facts["platform"]):
+                    combined_drop_counter = True
+                    break
+
+    setup_information = {
+        "port_channel_members": port_channel_members,
+        "vlan_members": vlan_members,
+        "rif_members": rif_members,
+        "dut_to_ptf_port_map": mg_facts["minigraph_port_indices"],
+        "combined_drop_counter": combined_drop_counter
+    }
+
+    return setup_information
+
+
+@pytest.fixture(params=["port_channel_members", "vlan_members", "rif_members"])
+def tx_dut_ports(request, setup):
+    """ Fixture for getting port members of specific port group """
+    return setup[request.param] if setup[request.param] else pytest.skip("No {} available".format(request.param))
+
+
+@pytest.fixture(autouse=True, scope="module")
+def enable_counters(duthost):
+    """ Fixture which enables RIF and L2 counters """
+    cmd_list = ["intfstat -D", "counterpoll port enable", "counterpoll rif enable", "sonic-clear counters",
```

Comment:
This fixture has to restore previous FC configuration

Reply:
Added restore functionality.